.github/workflows/test.yml

```diff
@@ -90,72 +93,29 @@ jobs:
     timeout-minutes: 60
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v4
-      - uses: actions/setup-node@v4
-        with:
-          node-version: 20.x
-      - name: Cache Node.js modules
-        uses: actions/cache@v4
+      - name: Setup Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v3
         with:
-          path: ~/.npm # npm cache files are stored in `~/.npm` on Linux/macOS
-          key: ${{ runner.OS }}-node-${{ hashFiles('**/package-lock.json') }}
-          restore-keys: |
-            ${{ runner.OS }}-node-
-            ${{ runner.OS }}-
-      - name: Install Packages
-        run: npm ci
-      - name: Build tsoa
-        run: npm run tsoa:build
-      - name: Build
-        run: npm run build
-      - name: touch env
-        run: touch .env
-      - name: Setup dependencies
-        run: docker compose up -d
-      - name: Sleep
-        uses: kibertoad/wait-action@1.0.1
+          buildkitd-flags: '--debug'
+      - uses: actions/checkout@v4
+      - name: Build e2e containers
+        uses: docker/bake-action@v5
         with:
-          time: '30s'
-      - name: Run database migrations
-        run: npm run db:migrate
+          builder: ${{ steps.buildx.outputs.name }}
+          files: ./docker-compose-test.yml
         env:
-          NODE_ENV: test
-      - name: Start server in detached mode
-        run: nohup npm run start &
+          VERITABLE_COMPANY_PROFILE_API_KEY: ${{ secrets.COMPANIES_HOUSE_API }}
+      - name: Run e2e tests
         env:
-          PORT: 3000
-          LOG_LEVEL: debug
-          DB_HOST: localhost
-          DB_NAME: veritable-ui
-          DB_USERNAME: postgres
-          DB_PASSWORD: postgres
-          DB_PORT: 5432
-          COOKIE_SESSION_KEYS: '["secret"]'
-          PUBLIC_URL: http://localhost:3000
-          IDP_CLIENT_ID: veritable-ui
-          IDP_PUBLIC_URL_PREFIX: http://localhost:3080/realms/veritable/protocol/openid-connect
-          IDP_INTERNAL_URL_PREFIX: http://localhost:3080/realms/veritable/protocol/openid-connect
-          CLOUDAGENT_ADMIN_ORIGIN: http://localhost:3100
-          CLOUDAGENT_ADMIN_WS_ORIGIN: ws://localhost:3100
-          INVITATION_PIN_SECRET: secret
-          INVITATION_FROM_COMPANY_NUMBER: 07964699
-          ISSUANCE_DID_POLICY: EXISTING_OR_NEW
-          ISSUANCE_SCHEMA_POLICY: EXISTING_OR_NEW
-          ISSUANCE_CRED_DEF_POLICY: EXISTING_OR_NEW
-          COMPANY_PROFILE_API_KEY: apikey
-          SMTP_HOST: localhost
-          SMTP_PASS: ''
-          SMTP_PORT: 2525
-          SMTP_USER: ''
-      - name: Install Playwright Browsers
-        run: npx playwright install --with-deps
-      - name: Run Playwright tests
-        run: npm run test:playwright
+          VERITABLE_COMPANY_PROFILE_API_KEY: ${{ secrets.COMPANIES_HOUSE_API }}
+          VERITABLE_E2E_OUT_DIR: /home/runner/work/_temp/
```

Comment:
you should be able to template this in as `${{ runner.temp }}`. Just in case github ever changes things